From 7b7e0307f65c10c40b12d9ccb6308b9cc1466e70 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Wed, 26 Jan 2022 01:23:21 +0200
Subject: [PATCH] [SW-67515] thermal: fix reg to c calc

Perform mC calculation as it is done in arcpid, removing the need of
multiplying by 10 by multiplying Greco's coefficients by 10 (same as in
zephyr Greco's dts).

Change-Id: I63b523fce15131b8408d59741b2ef398e5433493
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/arch/greco/arc/common.h b/arch/greco/arc/common.h
index e0877e7..9b0ffdb 100644
--- a/arch/greco/arc/common.h
+++ b/arch/greco/arc/common.h
@@ -57,9 +57,9 @@
 
 #define SHA256_ENABLE
 
-#define TS_UNCALIB_A_G_COEFF	5740
-#define TS_UNCALIB_B_H_COEFF	24940
-#define TS_CALIB_A_G_COEFF	6567
-#define TS_CALIB_B_H_COEFF	25410
+#define TS_UNCALIB_A_G_COEFF	57400
+#define TS_UNCALIB_B_H_COEFF	249400
+#define TS_CALIB_A_G_COEFF	65670
+#define TS_CALIB_B_H_COEFF	254100
 
 #endif /* COMMON_GRECO_ARC_GNU_H_ */
diff --git a/arch/greco/arm/common.h b/arch/greco/arm/common.h
index f1ae854..5ee44df 100644
--- a/arch/greco/arm/common.h
+++ b/arch/greco/arm/common.h
@@ -60,9 +60,9 @@
 
 #define SHA256_ENABLE
 
-#define TS_UNCALIB_A_G_COEFF	5740
-#define TS_UNCALIB_B_H_COEFF	24940
-#define TS_CALIB_A_G_COEFF	6567
-#define TS_CALIB_B_H_COEFF	25410
+#define TS_UNCALIB_A_G_COEFF	57400
+#define TS_UNCALIB_B_H_COEFF	249400
+#define TS_CALIB_A_G_COEFF	65670
+#define TS_CALIB_B_H_COEFF	254100
 
 #endif /* COMMON_GRECO_H_ */
diff --git a/lib/preboot/thermal_v2.c b/lib/preboot/thermal_v2.c
index ebd4fed..681ff3d 100644
--- a/lib/preboot/thermal_v2.c
+++ b/lib/preboot/thermal_v2.c
@@ -152,13 +152,6 @@
 	tmp = coef_a_g + ((coef_b_h * (((tmp * 1000) / coef_div) -
 					500)) / 1000);
 
-	/*
-	 * result of equation has to be multiplied
-	 * by 1000, since coefficients are multiplied by 100 an extra
-	 * multiplication of 10 is required.
-	 */
-	tmp *= 10;
-
 	return tmp;
 }
 
